### PR TITLE
[STRAT-3534] | Fixed Google Enhanced Conversion Action Dynamic Field Dropdown 

### PR DESCRIPTION
**Pull request by Innovative-GauravKochar**
This Pull request is to update version in Dynamic field Conversion Action of google-enhanced-conversions and removed hyphen from customerId to fix 400 Bad request.

Provided **Support for feature and StatsContext** in Dynamic Field. Related PR :- https://github.com/segmentio/integrations/pull/2807

Jira ticket :- https://segment.atlassian.net/browse/STRATCONN-3534

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

Screenshot 2024-01-23 at 5 20 40 PM

### CODE REVIEW

**Review by varadarajan-tw — APPROVED** [review]
LGTM. We might have to make changes to dest-act-service for passing features and statsContext.

File: packages/core/src/destination-kit/action.ts [hunk 1]
```diff
@@ -149,6 +149,8 @@ export interface ExecuteDynamicFieldInput<Settings, Payload, AudienceSettings =
   payload: Payload
   page?: string
   auth?: AuthTokens
+  features?: Features | undefined
```
Comment (dlasky) [inline packages/core/src/destination-kit/action.ts:152]: i think the otehr place where we've exposed this there were some notes about internal use only? would recommend checking and adding here as well if so.

Comment (Innovative-GauravKochar) [reply packages/core/src/destination-kit/action.ts:153]: @dlasky Kindly review this PR https://github.com/segmentio/integrations/pull/2807

**Review by nick-Ag — APPROVED** [review]
Nice 🔥 Thanks for adding this

**Comment by joe-ayoub-segment** [comment]
hi @Innovative-GauravKochar can you address the failing tests? Prob just need to rebase from main. 
I can't deploy otherwise.

**Comment by joe-ayoub-segment** [comment]
Hi @Innovative-GauravKochar PR deployed

**Pull request merged**